tests/raftstore_cases/test_prevote.rs

```diff
@@ -0,0 +1,216 @@
+use std::sync::{atomic::AtomicBool, mpsc, Arc};
+use std::time::Duration;
+
+use super::cluster::{Cluster, Simulator};
+use super::server::new_server_cluster;
+use super::transport_simulate::*;
+use super::util::*;
+use raft::eraftpb::MessageType;
+
+enum FailureType<'a> {
+    Partition(&'a [u64], &'a [u64]),
+    Reboot(&'a [u64]),
+}
+
+fn attach_prevote_notifiers<T: Simulator>(cluster: &Cluster<T>, peer: u64) -> mpsc::Receiver<()> {
+    // Setup a notifier
+    let (tx, rx) = mpsc::channel();
+    let response_notifier = Box::new(MessageTypeNotifier::new(
+        MessageType::MsgRequestPreVoteResponse,
+        tx.clone(),
+        Arc::from(AtomicBool::new(true)),
+    ));
+    let request_notifier = Box::new(MessageTypeNotifier::new(
+        MessageType::MsgRequestPreVote,
+        tx.clone(),
+        Arc::from(AtomicBool::new(true)),
+    ));
+
+    cluster
+        .sim
+        .write()
+        .unwrap()
+        .add_send_filter(peer, response_notifier);
+    cluster
+        .sim
+        .write()
+        .unwrap()
+        .add_send_filter(peer, request_notifier);
+
+    rx
+}
+
+// Validate that prevote is used in elections after partition or reboot of some nodes.
+fn test_prevote<T: Simulator>(
+    cluster: &mut Cluster<T>,
+    failure_type: FailureType,
+    leader_after_failure_id: impl Into<Option<u64>>,
```

Comment:
em, why do you use `impl Into` here? just using `Option<u64>` is not enough?

Reply:
seem it is a new feature for assigning Option, we can pass value instead of `Some(value)` now, cool.